parastell/invessel_build.py

```diff
@@ -256,6 +302,150 @@ def generate_components(self):
             self.Components[name] = component
             interior_surface = outer_surface
 
+    def connect_ribs_with_tris_moab(self, rib1, rib2, reverse=False):
+        """Creat MBTRI elements add add them to a surface between two ribs.
+
+        Arguments:
+            rib1 (Rib object): First of two ribs to be connected.
+            rib2 (Rib object): Second of two ribs to be connected.
+            reverse (bool): Optional. Whether to reverse the connectivity of
+                the MBTRIs being generated. Defaults to False.
+
+        Returns:
+            mb_tris (list of Entity Handle): List of the entity handles of the
+                MBTRIs connecting the two ribs.
+        """
+        mb_tris = []
+        for rib_loci_index, _ in enumerate(rib1.rib_loci[0:-1]):
+            corner1 = rib1.mb_verts[rib_loci_index]
+            corner2 = rib1.mb_verts[rib_loci_index + 1]
+            corner3 = rib2.mb_verts[rib_loci_index + 1]
+            corner4 = rib2.mb_verts[rib_loci_index]
+            corners = [corner1, corner2, corner3, corner4]
+            mb_tris += create_moab_tris_from_verts(
+                corners, self.mbc, reverse=reverse
+            )
+        return mb_tris
+
+    def generate_pymoab_verts(self):
+        """Generate MBVERTEX entities from rib loci in all surfaces"""
+        [
+            surface.generate_pymoab_verts(self.mbc)
+            for surface in self.Surfaces.values()
+        ]
+
+    def generate_curved_surfaces_pydagmc(self):
+        """Generate the faceted representation of each curved surface and
+        add it to the PyDAGMC model, remembering the surface ids."""
+        curved_surface_ids = []
+        for i, surface in enumerate(self.Surfaces.values()):
+            mb_tris = []
+            for rib, next_rib in zip(surface.Ribs[0:-1], surface.Ribs[1:]):
+                for rib_pt_index, _ in enumerate(rib.rib_loci[0:-1]):
+                    corner1 = rib.mb_verts[rib_pt_index]
+                    corner2 = rib.mb_verts[rib_pt_index + 1]
+                    corner3 = next_rib.mb_verts[rib_pt_index + 1]
+                    corner4 = next_rib.mb_verts[rib_pt_index]
+                    corners = [corner1, corner2, corner3, corner4]
+                    mb_tris += create_moab_tris_from_verts(
+                        corners,
+                        self.dag_model.mb,
+                        reverse=True if i == 0 else False,
+                    )
+            dagmc_surface = dagmc.Surface.create(self.dag_model)
```

Comment:
`DAGModel` provides an interface for this
```suggestion
            dagmc_surface = self.dag_model.create_surface()
```